### PR TITLE
ADEN-7891 Add wgVar to add rv and s1 to Moat FV tracking.

### PR DESCRIPTION
**Pull request by jacekbj**
I moved Moat tracking to FV tracking because that was the simplest way of passing `videoDepth` to Moat.

### CODE REVIEW

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 1]
```diff
@@ -150,6 +151,7 @@ public static function onInstantGlobalsGetNewsAndStoriesVariables( array &$vars
 		$vars[] = 'wgAdDriverVideoMoatTrackingSampling';
 		$vars[] = 'wgAdDriverVideoPostrollCountries';
 		$vars[] = 'wgAdDriverLABradorDfpKeyvals';
+		$vars[] = 'wgAdDriverMoatTrackingForFeaturedVideoRvS1Countries';
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: Not neeeded

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 2]
```diff
@@ -86,6 +86,7 @@ public static function onInstantGlobalsGetVariables( array &$vars ) {
 		$vars[] = 'wgAdDriverMegaAdUnitBuilderForFVCountries';
 		$vars[] = 'wgAdDriverMoatTrackingForFeaturedVideoAdCountries';
 		$vars[] = 'wgAdDriverMoatTrackingForFeaturedVideoAdSampling';
+		$vars[] = 'wgAdDriverMoatTrackingForFeaturedVideoRvS1Countries';
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: What about `wgAdDriverMoatTrackingAdditionalParametersCountries`?

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js [hunk 3]
```diff
@@ -54,6 +57,27 @@ define('wikia.articleVideo.featuredVideo.adsConfiguration', [
 		return {};
 	}
 
+	function loadMoatTrackingPlugin() {
```
Comment (fraszczakszymon) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js:60]: FYI: it's moved to ad-engine in ADEN-7818

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js [hunk 4]
```diff
@@ -54,6 +57,27 @@ define('wikia.articleVideo.featuredVideo.adsConfiguration', [
...
+		}
+	}
+
+	/**
+	 * Calculate depth
+	 *
+	 * @param {number} depth
+	 * @returns {number} rv
+	 */
+	function calculateRV(depth) {
```
Comment (fraszczakszymon) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js]: Can we reuse method from `articleVideoAd` module?

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js [hunk 5]
```diff
@@ -1,36 +0,0 @@
-define('wikia.articleVideo.featuredVideo.moatTracking', [
```
Comment (fraszczakszymon) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js]: It's used by mobile-wiki with AE2... I know, I know... 🙈

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php (hunk 2 above)
Comment (jacekbj) [reply extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: We have separate moat wgVars for Porvata, so what about `wgAdDriverMoatTrackingForFeaturedVideoAdditionalParamsCountries` :D?

EDIT
...or `wgAdDriverMoatTrackingFVAdditionalParamsCountries`

Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: I like this one: wgAdDriverMoatTrackingForFeaturedVideoAdditionalParamsCountries
😆

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js (hunk 5 above)
Comment (jacekbj) [reply extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js]: screen shot 2018-11-19 at 14 19 46
We will get there 💪 🚀 !

Comment (fraszczakszymon) [reply extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js]: Not really :(
https://github.com/Wikia/mobile-wiki/blob/dev/app/modules/ads/legacyModule.js#L134

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php (hunk 1 above)
Comment (jacekbj) [reply extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: Removed

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php (hunk 2 above)
Comment (jacekbj) [reply extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: Staying at wgAdDriverMoatTrackingForFeaturedVideoAdditionalParamsCountries

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js (hunk 4 above)
Comment (jacekbj) [reply extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js]: Refactored.

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js (hunk 5 above)
Comment (jacekbj) [reply extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js]: Please pay special attention to this part as I refactored it slightly to make it work with AE2.

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 6]
```diff
@@ -178,6 +179,7 @@ public static function onInstantGlobalsGetFandomCreatorVariables( array &$vars )
 		$vars[] = 'wgAdDriverVideoMidrollCountries';
 		$vars[] = 'wgAdDriverVideoMoatTrackingCountries';
 		$vars[] = 'wgAdDriverVideoMoatTrackingSampling';
+		$vars[] = 'wgAdDriverMoatTrackingForFeaturedVideoAdditionalParamsCountries';
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: This one should go to `onInstantGlobalsGetNewsAndStoriesVariables` list

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**